### ISSUE TITLE
Evaluation of the real and complex part of conductivity

### ISSUE THREAD

**Issue by algebrato** [FaustinCarter/barmat#2]
Is possible to evaluate the imaginary and real part of the conductivity?

**Comment by FaustinCarter** [comment]
This really depends on the specifics of your problem. The complex conductivity (as defined by Tinkham) involves a subset of the integrals used to calculate the surface impedance and those are only valid under some limiting conditions. Jiansong Gao's 2008 dissertation talks about the relationship between surface impedance and complex conductivity. The discussion starts in section 2.4. To summarize: if your material is thick (compared to the penetration depth) and satisfies the conditions of either the extreme anomalous limit or the local limit, then you can use either Eqs. 2.73 or 2.78 to equate the output of barmat to the complex conductivity defined by Tinkham (depending on which limit you are in). If your film is thin, then you probably shouldn't be using barmat, as it only uses the formulas for a thick film (although I'm slowly working on remedying this!).

I realize this isn't a very satisfying answer, but keep in mind that the Tinkham complex conductivity was introduced before the BCS theory was released; a footnote in the Glover/Tinkham paper (https://journals.aps.org/pr/pdf/10.1103/PhysRev.108.243) that introduces the idea of a complex conductivity mentions that BCS was published while they were finishing up their manuscript! Bascially, the complex conductivity idea is an extension of Pippard's non-local extension of London's theory, and BCS (which is what Mattis/Bardeen used to calculate surface impedance) only reduces to the Pippard conjecture under certain conditions. So the complex conductivity isn't really a physical, microscopic, property of the superconductor, but in some limiting cases, it is a useful idea that helps to simplify calculations.

TL;DR: barmat calculates the complex surface impedance (a physical property of the material). This surface impedance is only relatable to the complex conductivity (a remnant of a non-BCS attempt at explaining superconductors in RF fields) under certain conditions, which are outlined in Gao's thesis.

**Comment by FaustinCarter** [comment]
One of the To Do items in barmat's next version is to check for these limits and use the simplified expressions when possible to reduce calculation time. I suppose it would be easy to add an option that outputs the conductivity in addition to the impedance if it makes sense to do so.
